ltiauthenticator/lti11/auth.py

```diff
@@ -35,6 +36,24 @@ class LTI11Authenticator(Authenticator):
         """,
     )
 
+    username_key = Unicode(
+        None,
+        allow_none=True,
+        config=True,
+        help="""
+        Key present in LTI 1.1 launch request used to set the user's JupyterHub's username.
+        Common standard (vendor agnostic) options are:
+          - User's email address: lis_person_contact_email_primary
+          - Opaque user id: user_id
+        Your LMS (Canvas / Open EdX / Moodle / others) may provide additional
+        keys in the LTI 1.1 launch request that you can use to set the username. In most cases these
+        are prefixed with `custom_`.
+        When the default of 'None' is set, the following backwards compatible behavior is provided:
+          If `canvas_custom_user_id` is present, that is used. Otherwise,
```

Comment:
Let's mark the canvas_custom_user_id default as deprecated? We no longer need it to be hard coded since people who want this behavior can set `username_key` instead.

Reply:
Glad you mentioned this. Agree and will make the change. We could set the username_key to default to user_id since it's the best common denominator (including for private launches when things are opaque). I also thought of adding a list of Enums to ensure the username_key is valid for LTI 1.1 but would rather do that in a separate PR.

Reply:
Yeah, I agree re: setting default to `user_id`. Let's mark the `canvas_custom_user_id` as deprecated, emit a warning if the user is relying on that, and then change the default on next release?

Reply:
Mostly since end users will see this breaking change as 'my files are all gone! PANIC!' so want to make sure people have at least some warning